fgpyo/sequence.py

```diff
@@ -75,16 +75,15 @@ def reverse_complement(bases: str) -> str:
 def longest_hp_length(bases: str) -> int:
     """Calculates the length of the longest homopolymer in the input sequence."""
     max_hp = 0
-    for i in range(0, len(bases)):
+    i = 0
+    # NB: if we have found a homopolymer of length `max_hp`, then we do not need
+    # to examine the last `max_hp` bases since we'll never find a longer one.
+    while i < len(bases) - max_hp:
```

Comment:
If going for speed, why not also cache `len(bases)`?

Reply:
It’s O(1) so as fast as an access, so really isn’t beneficial.